# xGBoost

In [15]:

# Imports
import numpy as np
import pandas as pd
import pickle

from numpy import loadtxt
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

In [6]:
# Now I am going to load the LoanPrediction Dataset.
df = pd.read_csv('loan_prediction.csv')

# Now I will split the data into X and y
X = df.iloc[:,0:-1]
Y = df.iloc[:,-1]

In [7]:
# Now let's split the data into train and test sets.
seed = 7
test_size = 0.33
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=test_size, random_state=seed)

In [9]:
# Let's now fit the model on the train data.
model = XGBClassifier()
model.fit(X_train, y_train)

/home/troyegilbert/.local/lib/python3.9/site-packages/xgboost/sklearn.py:1146: UserWarning: The use of label encoder in XGBClassifier is deprecated and will be removed in a future release. To remove this warning, do the following: 1) Pass option use_label_encoder=False when constructing XGBClassifier object; and 2) Encode your labels (y) as integers starting with 0, i.e. 0, 1, 2, ..., [num_class - 1].
  warnings.warn(label_encoder_deprecation_msg, UserWarning)


[22:35:07] WARNING: ../src/learner.cc:1095: Starting in XGBoost 1.3.0, the default evaluation metric used with the objective 'binary:logistic' was changed from 'error' to 'logloss'. Explicitly set eval_metric if you'd like to restore the old behavior.


XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0, gpu_id=-1,
              importance_type='gain', interaction_constraints='',
              learning_rate=0.300000012, max_delta_step=0, max_depth=6,
              min_child_weight=1, missing=nan, monotone_constraints='()',
              n_estimators=100, n_jobs=4, num_parallel_tree=1, random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,
              tree_method='exact', validate_parameters=1, verbosity=None)

In [12]:
# Now we'll try making predictions
y_pred = model.predict(X_test)
predictions = [round(value) for value in y_pred]

# Evaluation of our predictions
accuracy = accuracy_score(y_test, predictions)
print('Accuracy: %.2f%%' % (accuracy * 100))

Accuracy: 71.92%


### We can see that the XGBoost model predicted well with an accuracy of 71.92%

In [19]:
# Now we are going to save our model.
pickle.dump(model, open('loan_prediction.pickle.dat', 'wb'))

In [22]:
# Let's try loading the model.
loaded_model = pickle.load(open('loan_prediction.pickle.dat', 'rb'))
y_pred = loaded_model.predict(X_test)